ding/framework/event_loop.py

```diff
@@ -0,0 +1,129 @@
+from collections import defaultdict
+from typing import Callable, Optional
+from concurrent.futures import ThreadPoolExecutor
+import fnmatch
+import logging
+
+
+class EventLoop:
+    loops = {}
+
+    def __init__(self, name: str = "default") -> None:
+        self._name = name
+        self._listeners = defaultdict(list)
+        self._thread_pool = ThreadPoolExecutor(max_workers=2)
+        self._exc = None
```

Comment:
`self._exception` can be easier to understand

Reply:
done